### PR TITLE
fix: Apply x:Bind when it's referencing an `x:Load`ed element

### LINKED ISSUE #16250: WASM: Element binding is failing when referencing an element loaded via x:Load
### Current behavior

If the following is true:
- `Element1` in XAML is using `x:Bind` to reference `Element2`'s property, 
- `Element2` is loaded via `x:Load` 
- `Element2` is initially not loaded

WinUI tracks the binding such that when `Element2` is loaded, the binding on `Element1` tracks the value changes.
On WASM, after `Element2` is loaded, the binding does not begin monitoring the changes

### Expected behavior

WASM behaves the same as WinUI

### How to reproduce it (as minimally and precisely as possible)

[UnoElementLoadBinding.zip](https://github.com/unoplatform/uno/files/14959160/UnoElementLoadBinding.zip)

1. Open attached solution
2. Set Window as the startup project
3. Compile and run:
image
4. Note Buttons 1-4 are enabled.
5. Click [Toggle Load] to load another button.
6. Note a new button appears and Button 1-4 are now disabled.
image
8. Click the new button and note that Buttons 1-4 are now enabled.
image
10. Close the app.
11. In Visual Studio, select the WASM head as the startup.
12. Compile and run.
13. Repeat the above steps, but note that the enabled state of Buttons 1-4 never change,
image

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #16250

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 1]
```diff
@@ -1153,6 +1154,70 @@ private void BuildCompiledBindings(IndentedStringBuilder writer)
 						writer.AppendLineIndented($"Owner = owner;");
 					}
 
+					using (writer.BlockInvariant($"void {bindingsInterfaceName}.NotifyXLoad(string name)"))
```
Comment (jeromelaban) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:1157]: Could you add a sourcegen test that shows the generated code for that scenario??

**Review by jeromelaban — APPROVED** [review]

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs (hunk 1 above)
Comment (Youssef1313) [reply src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:1157]: Yup sure. Will add one now.

**Review by MartinZikmund — APPROVED** [review]

**Review by ajpinedam — APPROVED** [review]

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.2

**Pull request merged**